pkg/kube/controllers/extendedjob/job_reconciler.go

```diff
@@ -178,24 +180,28 @@ func (r *ReconcileJob) persistOutput(ctx context.Context, instance *batchv1.Job,
 		if err != nil {
 			return errors.Wrap(err, "getting pod output")
 		}
-		// Put full job output in debug log
-		ctxlog.Debug(ctx, string(result))
+
+		// Create secret
+		secretName := ejob.Spec.Output.NamePrefix + c.Name
 
 		var data map[string]string
 		err = json.Unmarshal(result, &data)
 		if err != nil {
+			ctxlog.WithEvent(&ejob, "ExtendedJob").Errorf(ctx, "non-JSON output was emitted for container '%s', secret '%s' cannot be created", instance.GetName(), secretName)
 			return errors.Wrap(err, "invalid output format")
```

Comment:
I think `Errorf` returns an err, I would just use that one here, so we have the same message.
Also the line above, it could be invalid JSON, not just non-JSON ;)

Hm, when `persistOutput` fails we create another error.

Reply:
thanks